yt/visualization/plot_window.py

```diff
@@ -659,6 +667,10 @@ def __init__(self, *args, **kwargs):
         self._splat_color = kwargs.pop("splat_color", None)
         PlotWindow.__init__(self, *args, **kwargs)
 
+    @property
+    def data_is_valid(self):
+        return self._data_valid and self._frb and self._frb._data_valid
```

Comment:
I commented on this before and I don't think you responded - can you explain why we need both `plot._data_valid` and `_frb._data_valid`? I'd prefer to only store this state in one place if possible.

Reply:
I've updated the code removing `plot._data_valid`, let's see if the test pass.